### PR TITLE
WT-7509 Document dangers around checkpointing a targeted list of objects

### PR DESCRIPTION
**Pull request by keithbostic**
Add a warning about checkpointing a target set of objects.
Reorder the WT_SESSION transaction methods, right now the list is completely unordered.

@quchenhao, @kommiharibabu, let me know if you disagree with this decision.

There's a prebuilt version of the docs for your review. The important changes are:

[WT_SESSION transaction method restructuring](https://keithbostic.github.io/docs.build/struct_w_t___s_e_s_s_i_o_n.html)
[The Ref Guide warning for the 'target' configuration](https://keithbostic.github.io/docs.build/struct_w_t___s_e_s_s_i_o_n.html#a6550c9079198955c5071583941c85bbf)
[The User Guide warning](https://keithbostic.github.io/docs.build/checkpoint.html#checkpoint_target)

I had to restructure session initialization in `session_api.c`, but that was a mechanical change.

### CODE REVIEW

File: src/include/wiredtiger.in [hunk 1]
```diff
@@ -1961,6 +1948,27 @@ struct __wt_session {
 	 */
 	int __F(checkpoint)(WT_SESSION *session, const char *config);
 
+	/*!
+	 * Reset the snapshot.
```
Comment (quchenhao) [inline src/include/wiredtiger.in:1952]: Why move to here?

Comment (keithbostic) [reply src/include/wiredtiger.in:1952]: As far as I can tell, the order was random before; so, I split it up into 3 groups: "transactions" (so, begin, commit, prepare, rollback), "timestamps" (so, query_timestamp and timestamp_transaction), and "support" (checkpoint, reset_snapshot and transaction_pinned_range). I'm open to any ordering you like better, but what we had didn't make sense to me.

**Review by quchenhao — APPROVED** [review]
lgtm

**Comment by sauclovian-wt** [comment]
lgtm

**Pull request merged**